File: it.unibo.cautiousExplorer/app/src/main/java/it/unibo/cautiousExplorer/RobotMovesInfo.java
package it.unibo.cautiousExplorer;

import mapRoomKotlin.mapUtil;

import static mapRoomKotlin.mapUtil.*;
import static mapRoomKotlin.mapUtil.getMapRep;

public class RobotMovesInfo {
    private boolean  doMap = false;
    private String journey = "";

    public RobotMovesInfo(boolean doMap){
        this.doMap = doMap;
    }
    public void showRobotMovesRepresentation(  ){
        if( doMap ) showMap();
        else System.out.println( "journey=" + journey );
    }

    public String cleanMovesRepresentation(  ){
        if( doMap ) return getMapAndClean(); //getCleanMap();
        else {
            String answer = journey;
            journey       = "";
            return answer;
        }
    }

    public String getMovesRepresentation(  ){
        if( doMap ) return getMapRep();
        else return journey;
    }

    public void updateMovesRep(String move ){
        if( doMap )  doMove( move );
        else journey = journey + move;
    }


}

File: it.unibo.kotlinSupports/app/src/main/kotlin/it/unibo/actor0robot/RobotMovesInfo.kt
package it.unibo.actor0robot

import mapRoomKotlin.mapUtil.showMap
import mapRoomKotlin.mapUtil.getMapAndClean
import mapRoomKotlin.mapUtil.getMapRep
import mapRoomKotlin.mapUtil.doMove

class RobotMovesInfo(val doMap: Boolean) {

    private var journey = ""
    fun showRobotMovesRepresentation() {
        if (doMap) showMap() else println("journey=$journey")
    }

    fun cleanMovesRepresentation(): String {
        return if (doMap) getMapAndClean() //getCleanMap();
        else {
            val answer = journey
            journey = ""
            answer
        }
    }

    val movesRepresentation: String
        get() = if (doMap) getMapRep() else journey

    fun updateMovesRep(move: String) {
        if (doMap) doMove(move) else journey = journey + move
    }


}